File: Application/Services/Operations/Outsorced/Dtos/CollectDeliverDto.cs
using System;
using System.Collections.Generic;
using Application.Services.Operations.Authentication.Dtos;
using Application.Services.Operations.Main.Companies.Dtos;
using Application.Services.Operations.Main.Partners.Dtos;
using Application.Services.Shared.Dtos;

namespace Application.Services.Operations.Outsourced.Dtos
{
    public class CollectDeliverDto : RootBaseDto
    {
        public int? TransporterId { get; set; }
        public PartnerDto Transporter { get; set; }
        public string SubjectReason { get; set; }
        public string ContactName { get; set; }
        public DateTime Start { get; set; }
        public decimal Price { get; set; }
        public DateTime WasPaid { get; set; }
        public DateTime Collect { get; set; }
        public DateTime Deliver { get; set; }
        public DateTime Other { get; set; }
        public string KindTransport { get; set; }
        public BillingFromDto BillingFrom { get; set; }
        public string TaskOverView { get; set; }
        public DestinyDto Destiny { get; set; }
    }
}

File: Views/src/components/out-sourced/collect-deliver/dto/collect-deliver-dto.ts
import { RootBase } from "../../../../shared/entities-dtos/root-base";
import { PartnerDto } from "../../../../components/main/partner/dtos/partner-dto";
import { BillingFromDto } from "./billing-from-dto";
import { DestinyDto } from "./destiny-dto";

export class CollectDeliverDto extends RootBase {
  transporterId: number;
  transporter: PartnerDto;
  contactName: string;
  start: Date | string;
  price: number;

  collect: boolean | string | Date;
  deliver: boolean | string | Date;
  other: boolean | string | Date;

  wasPaid: Date | string;
  kindTransport: string;
  billingFromId: number;
  billingFrom: BillingFromDto;
  taskOverView: string;
  destinyId: number;
  destiny: DestinyDto;
}
